sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultStatementParameterProcessor.java

```diff
@@ -720,13 +706,32 @@ protected void setSmallInt(PreparedStatement preparedStatement, String sqlType,
     @Override
     protected void setFloat(PreparedStatement preparedStatement, String sqlType, int index, Object value)
             throws SQLException, ApplicationError {
-        setFloatAndReal(preparedStatement, sqlType, index, value);
+        if (value == null) {
```

Comment:
Is there any change to the logic? If not what is the use of duplicating the code?

Reply:
For Real type, added new condition 
``
else if (value instanceof Double) {
            preparedStatement.setDouble(index, ((Number) value).doubleValue());
}
``
and for Float type the conditions are kept as is.

Reply:
Ack.